superset/assets/spec/javascripts/dashboard/util/getFilterScopeFromNodesTree_spec.js

```diff
@@ -0,0 +1,216 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import getFilterScopeFromNodesTree from '../../../../src/dashboard/util/getFilterScopeFromNodesTree';
+
+describe('getFilterScopeFromNodesTree', () => {
+  it('should return empty scope', () => {
+    const nodes = [];
+    expect(
+      getFilterScopeFromNodesTree({
+        filterKey: '107_region',
+        nodes,
+        checkedChartIds: [],
+      }),
+    ).toEqual({});
+  });
+
+  it('should return scope for simple grid', () => {
+    const nodes = [
+      {
+        label: 'All dashboard',
+        type: 'ROOT',
+        value: 'ROOT_ID',
+        children: [
+          {
+            value: 104,
+            label: 'Life Expectancy VS Rural %',
+            type: 'CHART',
+          },
+          { value: 105, label: 'Rural Breakdown', type: 'CHART' },
+          {
+            value: 106,
+            label: "World's Pop Growth",
+            type: 'CHART',
+          },
+          {
+            label: 'Time Filter',
+            showCheckbox: false,
+            type: 'CHART',
+            value: 108,
+          },
+        ],
+      },
+    ];
+    const checkedChartIds = [104, 106];
+    expect(
+      getFilterScopeFromNodesTree({
+        filterKey: '108___time_range',
+        nodes,
+        checkedChartIds,
+      }),
+    ).toEqual({
+      scope: ['ROOT_ID'],
+      immune: [105],
```

Comment:
yes. For this test case, the filter apply to all charts under `ROOT_ID`, which has charts [104, 105, 106, 108]. chart 108 is filter_box itself, it will be immune automatically.